pkg/ignition/ignition.go

```diff
@@ -84,6 +84,15 @@ func (ign *Ignition) GetKubeletArgs() (map[string]string, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error parsing kubelet systemd unit args: %w", err)
 	}
+	// Special casing for vSphere to disable CSI Migration. These args have the potential to be either set to this, or
+	// not, depending on if the cluster was upgraded from 4.12 or if it is a fresh 4.13 install.
+	// See: https://github.com/openshift/machine-config-operator/pull/3655
+	// Because in-tree storage is the only option for Windows in 4.13 vSphere clusters, these values should always be
+	// set to this if the cluster platform is vSphere.
```

Comment:
Shouldn't we be looking at the storage CR like in https://github.com/openshift/machine-config-operator/pull/3655 instead always defaulting to using the cloud-provider option?

Reply:
IMO even when user opts-in for CSI migration on Linux nodes, Windows nodes should still use in-tree. Or are you going to require CSI driver to be installed on the Windows nodes too?

Reply:
@aravindhp if that was done we would have no way to support Windows storage for fresh 4.13 cluster installs in 8.0.0.